pkg/secrets/repositoryserver/secrets_interface.go

```diff
@@ -0,0 +1,5 @@
+package repositoryserver
+
+type RepositoryServerSecrets interface {
```

Comment:
```suggestion
type RepositoryServerSecret interface {
```
Since we will be running validation one secret at a time, could it be singular?